### DISCUSSION TITLE
Code 400 Bad Request when inserting data into table

### DISCUSSION THREAD

**Discussion by OthmaneHachad** [supabase/supabase#20593] [Questions]
I've been playing around with React Native coupled with Supabase lately. In this test app I need to save appointments into a table called "appointment". This is done in one of my components where data is fetched from either Global React Context or other tables from the supabase project. Here is the code (css excluded): 
  `
  const [client, setClient] = useState('');
    const [clientId, setClientId] = useState('');
    const [services, setServices] = useState<Service[] | undefined>(undefined);
    const [selectedService, setSelectedService] = useState('');

    const [date, setDate] = useState(new Date());
    const [showDatePicker, setShowDatePicker] = useState(false);

    const { workingAt } = useAuth();

    const [employeeBadges, setEmployeeBadges] = useState<EmployeeBadge[] | undefined>(undefined) ;
    const [selectedEmployee, setSelectedEmployee] = useState<EmployeeBadge>();

    // ... rest of the code

    const handleSubmit = async () => {
            await getClientId() ;
            //const employee_id = selectedEmployee?.id ;
            const { error } = await supabase
            .from('appointment')
            .insert(
              {
                date_time : date.toISOString(),
                service : selectedService,
                establishement : workingAt,
                employee : 'ff647f27-d06c-41be-9232-0ba8203a4250', 
                client : clientId 
              }
            );
    
            if (error) {
              Alert.alert("Error creating Appointment: ", error.message);
            } else {
              console.log('Appointment succesfully created !');
            }
        };
    
        return (
             // ... rest of the code
    
              <Button
                  title="Create Appointment"
                  onPress={handleSubmit}
                  buttonStyle={styles.button}
                  titleStyle={styles.buttonText}
              />
          </View>
      );
    };`

The issue lies in the request since it returns a code 400 (also getClientId() successfully fetches the client ID). Also note that all RLS policies have been setup correctly. 
    
Also documentation says this:
[4XX status codes](https://supabase.com/docs/guides/platform/http-status-codes#4xx-client-errors) indicate an issue on the client's end with the request being made. These could include missing or invalid auth information, a malformed request, making too many requests in too short a time period ("rate limiting"), or a network issue on the client's end.

**Where do you guys think I should be looking?**

**Comment by GaryAustin1** [comment]
What is the error message you get back?
If you look in the dashboard API Edge logs do you see the request making it to the server?
Just having a clientID does not mean you actually have a session in the Supabase client, but I would expect an error message for insert if RLS issue.

**Reply by OthmaneHachad** [reply]
Hey so the error message I found in the logs is in the image attached.
![image](https://github.com/supabase/supabase/assets/75754374/2c87c135-2d3c-4bfd-b25f-66737eceb744)
The clientID doesn't have anything to do anything with my session because the "client" isn't the one booking the appointment, its done on the business side. Also I made sure the type of user booking creating the appointment had a session as everytime I run the app I have to log in. Also I doubled checked to make sure the RLS were correct.

In addition, here is the code that tries to insert a row. 
![image](https://github.com/supabase/supabase/assets/75754374/e954275a-04cf-43c6-ba56-0913d3624229)

**Reply by GaryAustin1** [reply]
Sorry, this seems odd.  The error message you show implies missing query parameters but insert does not have query parameters.
Where are you seeing that actual error message (which logs)?
I don't find that error message with supabase on the web.

**Reply by OthmaneHachad** [reply]
I found it at reports/api-overview in the project dashboard. When I run the app and I look at the console in the developer mode of the browser I get the same error excluding the "no query parameters" part. It is only saying 400 bad Request. I don't get any other hint for where is the error from.

**Reply by GaryAustin1** [reply]
Ah.  That section was recently added.  The message there just means there were no query parameters in the call which is expected (my best guess) and is not the error message.  So not much help.
Try looking in the logs tab at api edge log and see if there is more info.
You seem to log out error.message.  Log out the entire error object and see if there is more info.

**Reply by nerded1337** [reply]
Hi, we are hitting the http 400 issue since yesterday while we didn't push any code to prod (the same code has been running smoothly since september with http 201 every time).
Any news on this?

**Reply by rokanost** [reply]
I'm also experiencing this issue from time to time

**Comment by tanakaa11** [comment]
I managed it fix this error, you just have mess around with your bucket policies. Mine worked when i added this code : 

create policy "Allow all inserts to product-images"
on storage.objects
for insert
to public
with check (
  bucket_id = 'your_bucket_name_here_xo'
  AND (auth.role() = 'authenticated' OR auth.role() = 'anon')
);

**Comment by Doseung-Yang** [comment]
Try disabling the RLS (Row-Level Security) policy in a test project with the same environment instead of your main (live) project.
If the issue is resolved when RLS is turned off, that indicates the policy should be reviewed again

Even if you believe “RLS is properly configured”, most insert failures in Supabase are actually caused by missing or incorrect INSERT permissions in the RLS policy

Use the Supabase SQL Editor to run queries and check the exact policy conditions.
Alternatively, go to the Supabase Policy tab and make sure the “Enable INSERT” toggle is turned on

Hope that solves the issue! :)
